lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -819,14 +819,33 @@ void AtenAddTensorOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
 // AtenSizeOp
 //===----------------------------------------------------------------------===//
 
+static FailureOr<BaseTensorType> traceKnownSizeTensorType(Value value) {
```

Comment:
Can we generalize this a bit and share it with aten.size.int?

Reply:
Updated the function to accept `Optional<int64_t>` for the dimension -- if the value is `None`, then the function checks whether all dimension's sizes are known.